make-release.sh

```diff
@@ -27,132 +67,185 @@ Usage: $0 --version [VERSION TO RELEASE]
-    (( BASE2=BASE2-1 ))
-    replaceFieldSed $playbookfile 'prod-prev-ver' "\"${BASE1}.${BASE2}\""
   fi
+  echo "[INFO] Updating to version: ${VERSION} is an increment for current version: ${OLDVERSION}."
 }
 
 replaceFieldSed()
 {
   YAMLFILE=$1
   updateName=$2
   updateVal=$3
-  # echo "[INFO] * ${YAMLFILE} ==> ${updateName}: ${updateVal}"
-  sed -i ${YAMLFILE} -r -e "s#(    $updateName: ).+#\1${updateVal}#"
+  echo "[INFO] Updating file: ${YAMLFILE} on branch: ${TARGET_BRANCH}: setting attribute: ${updateName}: ${updateVal}"
+  sed -i "${YAMLFILE}" -r -e "s#(    $updateName: ).+#\1${updateVal}#"
 }
 
-bump_version() {
-  CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
-
-  NEXTVERSION=$1
-  BUMP_BRANCH=$2
-
-  git checkout ${BUMP_BRANCH}
-
-  echo "[INFO] Update project version to ${NEXTVERSION}"
-  if updateYaml ${NEXTVERSION} && [[ ${docommit} -eq 1 ]]; then
-    COMMIT_MSG="chore: Bump to ${NEXTVERSION} in ${BUMP_BRANCH}"
-    git commit -s -m "${COMMIT_MSG}" $playbookfile
-    git pull origin "${BUMP_BRANCH}"
-
-    set +e
-    PUSH_TRY="$(git push origin "${BUMP_BRANCH}")"
-
-    # shellcheck disable=SC2181
-    if [[ $? -gt 0 ]] || [[ $PUSH_TRY == *"protected branch hook declined"* ]] || [[ $PUSH_TRY == *"Protected branch update"* ]]; then
-    PR_BRANCH=pr-${BUMP_BRANCH}-to-${NEXTVERSION}
-      # create pull request for master branch, as branch is restricted
-      git branch "${PR_BRANCH}"
-      git checkout "${PR_BRANCH}"
-      git pull origin "${PR_BRANCH}"
-      git push origin "${PR_BRANCH}"
-      lastCommitComment="$(git log -1 --pretty=%B)"
-      hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"
-    fi 
-    set -e
-  fi
-  git checkout ${CURRENT_BRANCH}
+versionUpdate() {
+  case ${TARGET_BRANCH} in
+      "release-${VERSION}")
+        # Update the version, defined in the antora-playbook-for-publication.yml file, in following keys:
+        #    branches: 7.32.x
+        yamlfile=antora-playbook-for-publication.yml
+        replaceFieldSed $yamlfile 'branches' "\"${MAJOR}.${MINOR}.x\""
+        ;;
+      "${MAJOR}.${MINOR}.x")
+        # Update the version, defined in the antora.yml file, in following keys:
+        #    prod-prev-ver-major: "6" [never changes]
+        #    prod-ver-major: "7" [never changes]
+        #    prod-prev-ver: "7.24" [always prod-ver - 1]
+        #    prod-ver: "7.25"
+        #    prod-ver-patch: "7.25.2"
+        # Major version upgrade is expected to fail.
+        yamlfile=antora.yml
+        # prod-ver should never go down, only up
+        versionIsIncremented
+        replaceFieldSed $yamlfile 'prod-ver-major' "\"${MAJOR}\""
+        replaceFieldSed $yamlfile 'prod-ver' "\"${MAJOR}.${MINOR}\""
+        replaceFieldSed $yamlfile 'prod-ver-patch' "\"${MAJOR}.${MINOR}.${PATCH}\""
+        replaceFieldSed $yamlfile 'prod-prev-ver' "\"${MAJOR}.$((MINOR - 1))\""
+        # Generate single-sourced docs
+        bash -c tools/checluster_docs_gen.sh
```

Comment:
why `bash -c tools/...` ? why not just `./tools/...` ?

Reply:
No good reason. Fixed.